### ISSUE TITLE
Compilation issue with default argument set to None, linux macos.

### ISSUE THREAD

**Issue by jeanlaroche** [serge-sans-paille/pythran#904]
The following code does not compile in linux or mac os due to the default value being None I believe.
Note: fails on master and on fix/default-arguments-aliasing
```
import numpy as np
#pythran export test(int,int)
def test(sr,N):
    C = test2(sr, N) # <- does not compile
    # C = test2(sr, N, 5) # <- does compile
    return C

def test2(sr, N,fmax=None):
    if fmax is None:
        fmax = float(sr) / 2
    A = np.arange(1.+N/2)
    B = np.arange(1.+fmax/2)
    return A[0] - B[0]
```

**Comment by serge-sans-paille** [comment]
I already tried to solve that kind of code, but failed... i'll try again :-)

**Comment by jeanlaroche** [comment]
Thanks! The main issue is that it's not really hard to work around the problem, what's hard is finding out where it comes from in a long function. As you know, the compilation errors are very hard to relate to the code being pythranized, so in my case, it takes a lot of investigation time to pinpoint the origin. After that, finding a workaround isn't usually too hard.

**Comment by serge-sans-paille** [comment]
Similar to #648

**Comment by serge-sans-paille** [comment]
Should be fixed by #939

**Comment by jeanlaroche** [comment]
I missed your comment. It is now fixed on master at 5646f43553e4e785635dbec9f689e3bec0aa2570
